allennlp/common/file_utils.py

```diff
@@ -144,3 +149,95 @@ def get_from_cache(url: str, cache_dir: str = None) -> str:
             logger.info("removing temp file %s", temp_file.name)
 
     return cache_path
+
+
+def get_file_extension(path: str, dot=True, lower: bool = True):
+    """ Returns the file extension, by default case-lowered and including the dot. """
+    ext = os.path.splitext(path)[1]
+    ext = ext if dot else ext[1:]
+    return ext.lower() if lower else ext
+
+
+class CompressedFileUtils:
+
+    SUPPORTED_FORMATS = {'.gz', '.zip'}
+    READ_MODE_CHOICES = {'t', 'rt', 'b', 'rb'}
+    DEFAULT_ENCODING = 'utf-8'
+
+    @staticmethod
+    def read(path: str, mode: str = 't', encoding: str = None,
+             file_format: str = None):
```

Comment:
I don't like this name, usually `read()` takes a file handle and returns its contents. As I understand it, this code returns a file handle. Accordingly, shouldn't it be called `open()`?

Also, can you add a return type for it?

Reply:
I called it ``read`` because ``open`` suggests it supports also writing mode. I was only interested in reading a compressed file, so I implemented only that. But actually I could make it support even writing modes with no effort and call it ``open``. I'll do that. Thank you

Reply:
IMO it's fine to call it `open` and have it not support writing mode.

Reply:
Fine, I'll just rename it then. 

It's not clear to me what should I put as return type. Maybe ``io.IOBase``?